src/utility/copy.cpp

```diff
@@ -43,8 +147,38 @@ DuckdbCopy(PlannedStmt *pstmt, const char *query_string, struct QueryEnvironment
 		return false;
 	}
 
+	int token_start_offset = FindQueryTokenOffset(query_string, TO);
+	duckdb::string rewritten_query_string;
+
+	if (copy_stmt->query) {
+		List *rewritten;
+		RawStmt *raw_stmt;
+		Query *query;
+
+		raw_stmt = makeNode(RawStmt);
+		raw_stmt->stmt = copy_stmt->query;
+		raw_stmt->stmt_location = pstmt->stmt_location;
+		raw_stmt->stmt_len = pstmt->stmt_len;
+
+		rewritten = pg_analyze_and_rewrite_fixedparams(raw_stmt, query_string, NULL, 0, NULL);
+		query = linitial_node(Query, rewritten);
+		rewritten_query_string = duckdb::StringUtil::Format("COPY (%s) %s;", pgduckdb_pg_get_querydef(query, false),
+		                                                    query_string + token_start_offset);
```

Comment:
How about we include the filename instead of verbatim copy pasting the end of the string starting at TO (which we first have to find with your new FindQueryTokenOffset function)

```suggestion
		rewritten_query_string = duckdb::StringUtil::Format("COPY (%s) TO %s;", pgduckdb_pg_get_querydef(query, false),
		                                                    quote_literal_cstr(copy_stmt->filename);
```

Reply:
We can't use lexer on PG16/PG17 so relaying now quoted filename string to give us location of OPTIONS part of query.